src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -368,7 +382,9 @@ public void run() {
             }
 
             if (response.getRedirectUrl() == null) {
-                if (!response.getAvailableTransports().contains("WebSockets")) {
+                if ((this.transportEnum == TransportEnum.ALL && !(response.getAvailableTransports().contains("WebSockets") || response.getAvailableTransports().contains("LongPolling"))) ||
+                        (this.transportEnum == TransportEnum.WEBSOCKETS && !response.getAvailableTransports().contains("WebSockets")) ||
+                        (this.transportEnum == TransportEnum.LONG_POLLING && !response.getAvailableTransports().contains("LongPolling"))) {
```

Comment:
What if `transportEnum == TransportEnum.ALL`, but a WebSocketTransport instance was passed to the HubConnection ctor?

Reply:
That's an internal API. Transport is private so users can only specify transports through the enum.